arangod/Aql/ModificationNodes.cpp

```diff
@@ -55,13 +55,10 @@ ModificationNode::ModificationNode(ExecutionPlan* plan, arangodb::velocypack::Sl
                            : true) {}
 
 /// @brief toVelocyPack
-void ModificationNode::toVelocyPackHelper(VPackBuilder& builder, unsigned flags,
-                                          std::unordered_set<ExecutionNode const*>& seen) const {
-  // call base class method
-  ExecutionNode::toVelocyPackHelperGeneric(builder, flags, seen);
-
+void ModificationNode::doToVelocyPack(VPackBuilder& builder, unsigned flags) const {
   // add collection information
   CollectionAccessingNode::toVelocyPack(builder, flags);
+  CollectionAccessingNode::toVelocyPackHelperPrimaryIndex(builder);
```

Comment:
Adding the `toVelocyPackHelperPrimaryIndex` here is a change compared to the previous plan output.
Previously, the `toVelocyPackHelperPrimaryIndex` method was called when serializing REMOVE, UPDATE, REPLACE and UPSERT, but it was not called for INSERT.
I don't know if it does any harm to add this for INSERT nodes as well, just wondering.

Reply:
Previously every derived node called `ModificationNode::toVelocyPackHelperPrimaryIndex`. This is now streamlined by moving this call here. This actually allow me to remove the overrides in `UpdateNode` and `ReplaceNode` since these don't add any additional data.

Reply:
I should have read the comment more carefully..! Your are right, the `InsertNode` previously did not call `toVelocyPackHelperPrimaryIndex`, while the new implementation _does_ call it, so this is indeed a change in behavior.
`toVelocyPackHelperPrimaryIndex` adds a field `index` with an array containing information about the primary index. However, I don't even know if or where this information is used. At least loading the nodes from VPack does not read this information.

Reply:
i tend to agree @mpoeter here.
The "indexes" seems to be never read by any of the Nodes, also I have not seen any access to "indexes" in the RestHandlers (just Collection Methods)

The GraphNode seems to generate an "indexes" field as well, which is never read. Only the IndexNode seems to read it.

At this Code position it can also not be there for Backwards Compatibility, as this is coordinators, and they are upgrade last, so they do not talk to older servers.

Reply:
It looks like these "indexes" are not used anywhere and can be removed entirely. However, we decided to keep the current version for now and address this in a separate PR.